engines/director/graphics.cpp

```diff
@@ -565,13 +571,24 @@ void DirectorPlotData::inkBlitSurface(Common::Rect &srcRect, const Graphics::Sur
 	if (sprite == kTextSprite)
 		applyColor = false;
 
+	Common::Rect srfClip = srf->getBounds();
+
 	srcPoint.y = abs(srcRect.top - destRect.top);
 	for (int i = 0; i < destRect.height(); i++, srcPoint.y++) {
 		if (d->_wm->_pixelformat.bytesPerPixel == 1) {
 			srcPoint.x = abs(srcRect.left - destRect.left);
 			const byte *msk = mask ? (const byte *)mask->getBasePtr(srcPoint.x, srcPoint.y) : nullptr;
 
 			for (int j = 0; j < destRect.width(); j++, srcPoint.x++) {
+				if (!srfClip.contains(srcPoint)) {
```

Comment:
I think the execution will be abruptly stopped here and other points not processed. How the original behave here? Perhaps, it clips the result?

Reply:
inkBlitSurface I think is meant to be for exactly sized buffers without stretch, so the original code would eventually go out of bounds and segfault/trigger ASAN.  You're right about it stopping abruptly, I'll change it so it clips the result instead. Main idea is that we avoid a crash + get a warning if it's called incorrectly.

Reply:
The original code would go out of bounds and throw a segfault/ASAN error; I gather inkBlitSurface is meant to receive exactly matching extents for the copy. Main idea was to make it a warning instead of a crash. Clipping the result is a better idea, updated to do that.